sdk/storage/Azure.Storage.Common/src/GeoRedundantReadMode.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage
+{
+    /// <summary>
+    /// Pattern for retrying a read request between geo-redundant primary and secondary endpoints.
+    /// </summary>
+    public enum GeoRedundantReadMode
+    {
+        /// <summary>
+        /// Altnernate between primary and secondary endpoints, attempting primary first.
+        /// </summary>
+        PrimaryThenSecondary = 0,
+
+        /// <summary>
+        /// Altnernate between primary and secondary endpoints, attempting secondary first.
+        /// </summary>
+        SecondaryThenPrimary = 1,
+
+        /// <summary>
+        /// Only target the primary endpoint.
+        /// </summary>
+        PrimaryOnly = 2,
```

Comment:
What would be the point of setting PrimaryOnly? Isn't this equivalent to leaving the secondary Uri null?

Reply:
Really just a parity inclusion. Not needed. Technically, secondary only isn't needed either, as the clients could just be instantiated with secondary from the start. We could get away with just the first two enum values.

They were both included as the original feature request was to port the enum and its functionality from v11, which had these values. But also v11 had a full `StorageUri` class that contained primary and secondary together, so it made a bit more sense back then.

Reply:
Discussed separately with team. Will be removing PrimaryOnly and SecondaryOnly modes, as they are possible by just creating the client with the desired host. Adding support later, if necessary, would not be a breaking change.